happi/backends/multi_db.py

```diff
@@ -0,0 +1,193 @@
+"""
+Backend implementation that combines multiple backends.
+"""
+import functools
+import logging
+import re
+from typing import Any, Dict, List, Optional, Union
+
+from .core import ItemMeta, ItemMetaGen, _Backend
+
+logger = logging.getLogger(__name__)
+
+
+def prevent_duplicate_ids(fn):
+    """
+    Decorator that remembers which document _id's have been yielded and
+    prevents subsequent documents with matching _id's from being yielded
```

Comment:
Does this mean that the only way to "shadow"/replace an entry from another database is to cautiously make sure that the _id field matches exactly? Is there any way to do that in most of these backends? In the json backend you can hand-modify your file carefully I suppose.

It's worth noting that e.g. the "name" field doesn't have to match the "_id" field for happi to work.

Reply:
Your understanding is right.  I wrote this thinking that `_id` was the unique field, is that not the case?  Or is it that the json-backend needs the `_id` to match the top level key?

Reply:
`_id` is the unique field but I don't quite remember if there's a way to control the value of it in every case

Reply:
For example, what if I have a central mongodb + my personal little json file and I want to either shadow the existing items or I want to make sure I don't have an _id conflict- can I necessarily find a way to do these things for any backend?

Reply:
I see, `_id` isn't really a field the user has direct control over while saving, etc.  

Maybe shadowing by name is a better choice here?

Reply:
I think I want to take a moment to read the code carefully and see who/when decides the id- this comment was triggered by checking our db and seeing that the ids and names don't quite match

Reply:
I think we need to chat about `_id` one day soon. I'd rather not get into the details on this PR, though.

Reply:
I suspect maybe we set the id to the name and then never update it if the name changes which might be the more appropriate thing to address rather than the implementation here
https://github.com/pcdshub/device_config/blob/e9fcbc89e03c5b0e2ac20e6633c6f2ee726a1007/db.json#L3-L23

Reply:
I think this PR does it correctly and I think there's some other bug in the code related to _id that let's our database look as it does for some entry